src/main/java/org/folio/service/exchangerate/ExchangeRateSourceService.java

```diff
@@ -0,0 +1,78 @@
+import org.folio.dao.exchangerate.ExchangeRateSourceDAO;
+import org.folio.rest.core.model.RequestContext;
+import org.folio.rest.exception.HttpException;
+import org.folio.rest.jaxrs.model.ExchangeRateSource;
+import org.folio.rest.util.ErrorCodes;
+
+import io.vertx.core.Future;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@AllArgsConstructor
+public class ExchangeRateSourceService {
+
+  private ExchangeRateSourceDAO exchangeRateSourceDAO;
+
+  public Future<ExchangeRateSource> getExchangeRateSource(RequestContext requestContext) {
+    var dbClient = requestContext.toDBClient();
+    return dbClient.withConn(conn -> exchangeRateSourceDAO.getExchangeRateSource(conn))
+      .map(exchangeRateSource -> exchangeRateSource.orElseThrow(() ->
+        new HttpException(NOT_FOUND.getStatusCode(), NOT_FOUND.getReasonPhrase())));
+  }
+
+  public Future<ExchangeRateSource> saveExchangeRateSource(ExchangeRateSource exchangeRateSource, RequestContext requestContext) {
+    if (!validateExchangeRateSource(exchangeRateSource)) {
+      return Future.failedFuture(new HttpException(422, ErrorCodes.EXCHANGE_RATE_SOURCE_INVALID.toError()));
+    }
+    return requestContext.toDBClient()
+      .withTrans(conn -> exchangeRateSourceDAO.saveExchangeRateSource(exchangeRateSource, conn))
+      .recover(this::handleException)
+      .onSuccess(v -> log.info("saveExchangeRateSource:: New exchange rate source was saved successfully"))
+      .onFailure(t -> log.error("Failed to save exchange rate source", t));
+  }
+
+  public Future<Void> updateExchangeRateSource(String id, ExchangeRateSource exchangeRateSource, RequestContext requestContext) {
+    if (!validateExchangeRateSource(exchangeRateSource)) {
+      return Future.failedFuture(new HttpException(422, ErrorCodes.EXCHANGE_RATE_SOURCE_INVALID.toError()));
+    }
+    return requestContext.toDBClient()
+      .withTrans(conn -> exchangeRateSourceDAO.updateExchangeRateSource(id, exchangeRateSource, conn))
+      .recover(this::handleException)
+      .onSuccess(v -> log.info("updateExchangeRateSource:: Exchange rate source with id: '{}' was updated successfully", id))
+      .onFailure(t -> log.error("Failed to update exchange rate source with id: '{}'", id, t));
+  }
+
+  public Future<Void> deleteExchangeRateSource(String id, RequestContext requestContext) {
+    var dbClient = requestContext.toDBClient();
+    return dbClient.withTrans(conn -> exchangeRateSourceDAO.deleteExchangeRateSource(id, conn))
+      .recover(this::handleException)
+      .onSuccess(v -> log.info("deleteExchangeRateSource:: Exchange rate source with id: '{}' was deleted successfully", id))
+      .onFailure(t -> log.error("Failed to delete exchange rate source with id: '{}'", id, t));
+  }
+
+  private boolean validateExchangeRateSource(ExchangeRateSource exchangeRateSource) {
+    return exchangeRateSource != null && StringUtils.isNotEmpty(exchangeRateSource.getProviderUri());
```

Comment:
api key also should be required field, so we should remain it here in validation method. It is service so in future we will invoke some secure store instead of DAO, but validation to this field should exist in service method.